tests/pytests/unit/states/test_service.py

```diff
@@ -2,45 +2,522 @@
     :codeauthor: Gareth J. Greenaway <ggreenaway@vmware.com>
 """
 
+import logging
 import os
 
 import pytest
 import salt.modules.beacons as beaconmod
 import salt.states.beacon as beaconstate
-import salt.states.service as servicestate
+import salt.states.service as service
+import salt.utils.platform
 from salt.utils.event import SaltEvent
 from tests.support.mock import MagicMock, patch
 
+log = logging.getLogger(__name__)
+
+
+def func(name):
```

Comment:
does it make sense to use this function as a fixture instead?

Reply:
Not as familiar with the pytest fixtures, what would the advantage be?

Reply:
Since you access `func` only once from `mod_watch` function, you might pass just a return value from this fixture to `mod_watch` only, I don't know how stack looks like while using fixture vs function though, @s0undt3ch might provide more details on that.
Something like this should work for this test case:
```
@@ -15,12 +15,11 @@ from tests.support.mock import MagicMock, patch
 
 log = logging.getLogger(__name__)
 
-
-def func(name):
-    """
-        Mock func method
-    """
-    return name
+@pytest.fixture
+def func():
+    def request(name):
+        return name
+    return request
 
 
 @pytest.fixture
@@ -476,7 +475,7 @@ def test_disabled():
         assert service.__context__ == {"service.state": "disabled"}
 
 
-def test_mod_watch():
+def test_mod_watch(func):
     """
     Test to the service watcher, called to invoke the watch command.
     """
```